lib/td/client/api.rb

```diff
@@ -300,17 +301,17 @@ def do_get(url, params=nil, &block)
         else
           response = client.get(target, params, header)
 
-          validate_content_length!(response, response.body.bytesize) if @ssl
-        end
+          status = response.code
+          # retry if the HTTP error code is 500 or higher and we did not run out of retrying attempts
+          if !block_given? && status >= 500 && cumul_retry_delay < @max_cumul_retry_delay
+            $stderr.puts "Error #{status}: #{get_error(response)}. Retrying after #{retry_delay} seconds..."
+            sleep retry_delay
+            cumul_retry_delay += retry_delay
+            retry_delay *= 2
+            redo # restart from beginning of do-while loop
```

Comment:
Original code had `redo` out of if-else. Now it only exists in `else`.
Thus don't retry if block given and response code is 5xx. Is this behavior change intentional?

Reply:
Original code also had `redo` inside `if !block_given?`, isn't it?

Why not retry if block given is because if block given the block may be not idempotent.

Reply:
Ah I meant if-else for `if block ... else ...`
https://github.com/treasure-data/td-client-ruby/blob/e21f4d4681d8cae5712011c2acdce748a5f61651/lib/td/client/api.rb#L289-L314

```ruby
if block
  ...
else
  ...
end

if !block_given? && ..
  ...
  redo
end
```



changed to

```ruby
if block
  ...
else
  ...
  if !block_given? && ...
    ...
    redo
  end
end
```

Reply:
Ah nvm, sorry I got wrong about it. Behavior didn't changed